v8.go

```diff
@@ -77,14 +77,19 @@ type Snapshot struct{ data C.StartupData }
 
 func newSnapshot(data C.StartupData) *Snapshot {
 	s := &Snapshot{data}
-	runtime.SetFinalizer(s, (*Snapshot).release)
```

Comment:
Thanks for the new tests.

I went through this carefully last night.  We'll have to expose the release functions.  I was really hoping to avoid that -- I actually had explicit releasing in the code before I refactored and published the first version.  It makes tests much uglier because you need to release the Isolate and the Context for each test.  In my experience, most usage of the contexts and isolates are long-lived (e.g. as pools like you mentioned), so figuring out when to release the resources can be annoying.  I'd really like for the context to be able to clean itself up, but I don't see a clean way of doing that with the multiple reference cycles that we have between the map, the Values, and the Context itself.  Darn!

As for depending on the finalizers, I've gone back and forth between them.  I was convinced that it makes sense to use them to tie C memory to Go objects from this thread: https://groups.google.com/d/msg/golang-dev/DMiUkpS1uyQ/0sHpTacG00YJ (in particular the post from rsc about whether they will actually run).